src/pages/permissions/actions/DeleteIdpGroupsModal.tsx

```diff
@@ -0,0 +1,72 @@
+import { ConfirmationModal, useNotify } from "@canonical/react-components";
+import { useQueryClient } from "@tanstack/react-query";
+import { deleteIdpGroup, deleteIdpGroups } from "api/auth-idp-groups";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { FC, useState } from "react";
+import { IdpGroup } from "types/permissions";
+import { pluralize } from "util/instanceBulkActions";
+import { queryKeys } from "util/queryKeys";
+
+interface Props {
+  idpGroups: IdpGroup[];
+  close: () => void;
+}
+
+const DeleteIdpGroupsModal: FC<Props> = ({ idpGroups, close }) => {
+  const queryClient = useQueryClient();
+  const notify = useNotify();
+  const toastNotify = useToastNotification();
+  const [submitting, setSubmitting] = useState(false);
+  const hasOneGroup = idpGroups.length === 1;
+
+  const handleDeleteIdpGroups = () => {
+    setSubmitting(true);
+    const mutationPromise = hasOneGroup
+      ? deleteIdpGroup(idpGroups[0].name)
+      : deleteIdpGroups(idpGroups.map((group) => group.name));
+
+    const successMessage = hasOneGroup
+      ? `IDP group ${idpGroups[0].name} deleted.`
+      : `${idpGroups.length} IDP groups deleted.`;
+
+    mutationPromise
+      .then(() => {
+        void queryClient.invalidateQueries({
+          queryKey: [queryKeys.idpGroups],
+        });
+        toastNotify.success(successMessage);
+        close();
+      })
+      .catch((e) => {
+        notify.failure(
+          `${pluralize("IDP group", idpGroups.length)} deletion failed`,
+          e,
+        );
+      })
+      .finally(() => {
+        setSubmitting(false);
+      });
+  };
+
+  return (
+    <ConfirmationModal
+      confirmButtonLabel={`Permanently delete ${idpGroups.length} ${pluralize("IDP group", idpGroups.length)}`}
```

Comment:
The modal spacing is a bit odd and the confirm button copy seems too long

![image](https://github.com/canonical/lxd-ui/assets/1155472/b7f6ee38-0294-47ca-b03e-ece60b528691)

Reply:
Cool will go with the design which solves these issues. I was trying to align this one with the delete modal for lxd groups.